<a href="https://colab.research.google.com/github/Ori-Shoffet/Top-Movies/blob/main/Movies_Jupyter.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

This dataset was an easy decision because I love watching movies. I consider myself a movie buff and I knew that this Keggle Dataset would be an outlet for me to find new and great movies. 


Questions:
What movie descriptions are similar to each other?

How close are the most popular movies in voter count?

What is the earliest release dated movie to make this list?

How many movies made the popular list with the same leading actor or actress?

In [ ]:
import pandas as pd
import plotly.express as px
import matplotlib as plt

In [ ]:
df = pd.read_csv('https://github.com/Ori-Shoffet/Top-Movies/blob/main/Top%20Rated%20Movie%20Database.csv?raw=true')
df

Popularity  ...  Release Date
0        453.361  ...    2019-09-17
1        315.622  ...    2020-01-15
2        241.513  ...    2020-03-05
3        209.518  ...    2020-02-05
4        188.203  ...    2019-12-18
...          ...  ...           ...
9995       8.311  ...    2009-09-24
9996      11.059  ...    2002-07-13
9997       9.372  ...    2019-09-30
9998       9.061  ...    2007-09-28
9999       7.576  ...    2017-09-01

[10000 rows x 6 columns]

In [ ]:
df.shape

(10000, 6)

In [ ]:
df.info

<bound method DataFrame.info of       Popularity  ...  Release Date
0        453.361  ...    2019-09-17
1        315.622  ...    2020-01-15
2        241.513  ...    2020-03-05
3        209.518  ...    2020-02-05
4        188.203  ...    2019-12-18
...          ...  ...           ...
9995       8.311  ...    2009-09-24
9996      11.059  ...    2002-07-13
9997       9.372  ...    2019-09-30
9998       9.061  ...    2007-09-28
9999       7.576  ...    2017-09-01

[10000 rows x 6 columns]>

In [ ]:
df.head()

Popularity  ...  Release Date
0     453.361  ...    2019-09-17
1     315.622  ...    2020-01-15
2     241.513  ...    2020-03-05
3     209.518  ...    2020-02-05
4     188.203  ...    2019-12-18

[5 rows x 6 columns]

In [ ]:
df.tail()

Popularity  ...  Release Date
9995       8.311  ...    2009-09-24
9996      11.059  ...    2002-07-13
9997       9.372  ...    2019-09-30
9998       9.061  ...    2007-09-28
9999       7.576  ...    2017-09-01

[5 rows x 6 columns]

In [ ]:
df.sample(6)

Popularity  ...  Release Date
2395      15.877  ...    2010-06-14
7733       8.049  ...    2018-09-30
8392      11.648  ...    2009-06-26
9843       8.110  ...    1993-10-01
718       22.730  ...    2020-03-27
5962      11.914  ...    2015-07-10

[6 rows x 6 columns]

Let's Fix A Typo

In [ ]:
df.rename(columns={'Titile':'Title'})

Popularity  ...  Release Date
0        453.361  ...    2019-09-17
1        315.622  ...    2020-01-15
2        241.513  ...    2020-03-05
3        209.518  ...    2020-02-05
4        188.203  ...    2019-12-18
...          ...  ...           ...
9995       8.311  ...    2009-09-24
9996      11.059  ...    2002-07-13
9997       9.372  ...    2019-09-30
9998       9.061  ...    2007-09-28
9999       7.576  ...    2017-09-01

[10000 rows x 6 columns]

In [ ]:
df = df.rename(columns={'Titile':'Title'})
df

Popularity  ...  Release Date
0        453.361  ...    2019-09-17
1        315.622  ...    2020-01-15
2        241.513  ...    2020-03-05
3        209.518  ...    2020-02-05
4        188.203  ...    2019-12-18
...          ...  ...           ...
9995       8.311  ...    2009-09-24
9996      11.059  ...    2002-07-13
9997       9.372  ...    2019-09-30
9998       9.061  ...    2007-09-28
9999       7.576  ...    2017-09-01

[10000 rows x 6 columns]

Since we can't see the full overview, I say we get rid of it to save us from the confusion

In [ ]:
df_dropped = df.drop(columns={'Overview'})
df_dropped

Popularity  Vote Count  ... Vote Average  Release Date
9915       5.258          60  ...          6.9    1970-11-05
9541       5.387          88  ...          7.1    1943-08-05
9439       5.421          51  ...          6.0    2009-01-01
9801       5.424          95  ...          6.6    2001-11-06
9350       5.444          86  ...          6.6    1992-09-18
...          ...         ...  ...          ...           ...
5        192.374        2130  ...          7.4    2020-02-12
3        209.518        2717  ...          7.1    2020-02-05
2        241.513        1419  ...          7.2    2020-03-05
1        315.622        2653  ...          7.1    2020-01-15
0        453.361        2875  ...          5.9    2019-09-17

[10000 rows x 5 columns]

What could possibly be the 10 worst movies on a list of Top 10,000 Movies. Let's find out!

In [ ]:
df.sort_values(by = 'Popularity', inplace = True, ascending = True)
df

Popularity  ...  Release Date
9915       5.258  ...    1970-11-05
9541       5.387  ...    1943-08-05
9439       5.421  ...    2009-01-01
9801       5.424  ...    2001-11-06
9350       5.444  ...    1992-09-18
...          ...  ...           ...
5        192.374  ...    2020-02-12
3        209.518  ...    2020-02-05
2        241.513  ...    2020-03-05
1        315.622  ...    2020-01-15
0        453.361  ...    2019-09-17

[10000 rows x 6 columns]

In [ ]:
df_groupby = df.groupby('Vote Count',as_index=False).agg({'Vote Average':'mean'})
df_groupby

Vote Count  Vote Average
0              0      0.000000
1              1      6.634615
2              2      6.560714
3              3      6.461905
4              4      6.180000
...          ...           ...
2763       21588      8.400000
2764       21624      8.300000
2765       21923      7.700000
2766       22265      7.600000
2767       25168      8.300000

[2768 rows x 2 columns]

In [ ]:
df_filtered = df.head(100)
df_filtered

Popularity  ...  Release Date
9915       5.258  ...    1970-11-05
9541       5.387  ...    1943-08-05
9439       5.421  ...    2009-01-01
9801       5.424  ...    2001-11-06
9350       5.444  ...    1992-09-18
...          ...  ...           ...
8789       6.416  ...    2014-11-07
8648       6.422  ...    1963-09-23
6457       6.435  ...    1979-03-16
8518       6.435  ...    2003-02-08
8546       6.436  ...    2012-09-21

[100 rows x 6 columns]

In [ ]:
fig = px.scatter(df_filtered,title='Top Fifty Movies Voter Count Scatter Plot',y='Vote Count',x='Title', width=1000, height=1000)
fig.show()

In [ ]:
fig = px.bar(df_filtered[df_filtered['Vote Count'] > 100],title='Top Forty Movies by Voter Count Box Plot',x='Title',y='Vote Count',width=1000,height=1000,orientation='v')
fig.show()

In [ ]:
fig = px.box(df_filtered[df_filtered['Vote Average'] > 3],title='Average Vote Count Boxplot',x='Vote Average',orientation='h')
fig.show()

In [ ]:
fig = px.histogram(df_filtered,title='Popularity Rating Histogram',x='Popularity')
fig.show()

In [ ]:
fig = px.density_heatmap(df_filtered,title='Average Vote Per Movie Heatmap', x='Title',y='Vote Average',height=700)
fig.show()

In [ ]:
fig = px.scatter_3d(df_filtered,x='Vote Count',y='Vote Average',z='Title',color='Vote Count',title='3D Scatter',width=1000,height=1000)
fig.show()

In [ ]:
x = df['Popularity'].sample(1000,replace=True)
x.mean()

12.988069999999992

In [ ]:
x.std()

9.337375493621526

In a random sample of 1000 movies, the average popularity rating is 12.98. The average movie is 9.34 is above or below the average of these sampled movies. 

Conclusion: I chose this dataset to familiarize myself with top movies that I have not seen yet. Although there was an abundance of choices in my visualization packages, I was able to create a list of movies that I plan on watching, that I haven't already. The top 5 movies are Tigertail, Red River, The Hummingbird Project, Undercover Brother, and The Big Boss. I was caught between choosing this dataset and one pertaining to Top Songs on Billboards because I am as much of an avid listner to music as I am to watching movies. 